sdlib/d/gc/region.d

```diff
@@ -213,7 +209,7 @@ private:
 		maxAddress = max(maxAddress, v + blocks * BlockSize);
 
 		// Newly allocated blocks are considered clean.
-		return r.atClean(ptr, size);
+		return r.atClean(ptr, cast(uint) blocks);
```

Comment:
Add an assert that the value will fit on a uint.

You can do `blocks & uint.max` . VRP will do the job, and the operation that's actually performed is made explicit now.